File: docs/snippets/go/cli/cli_with_config.go
```go title="cli_with_config.go"
package main

import (
	"encoding/json"
	"fmt"
	"os/exec"
)

type ExtractedDocument struct {
	Content   string   `json:"content"`
	Format    string   `json:"format"`
	Languages []string `json:"languages"`
}

func extractWithConfig(filePath string, configPath string) (*ExtractedDocument, error) {
	cmd := exec.Command(
		"xberg",
		"extract",
		filePath,
		"--config",
		configPath,
		"--format",
		"json",
	)

	output, err := cmd.CombinedOutput()
	if err != nil {
		return nil, fmt.Errorf("CLI error: %w, output: %s", err, string(output))
	}

	var result ExtractedDocument
	if err := json.Unmarshal(output, &result); err != nil {
		return nil, fmt.Errorf("JSON parse error: %w", err)
	}

	return &result, nil
}

func main() {
	configFile := "xberg.toml"
	document := "document.pdf"

	fmt.Printf("Extracting %s with config %s\n", document, configFile)
	result, err := extractWithConfig(document, configFile)
	if err != nil {
		panic(err)
	}

	fmt.Printf("Content length: %d\n", len(result.Content))
	fmt.Printf("Format: %s\n", result.Format)
	fmt.Printf("Languages: %v\n", result.Languages)
}
```

File: docs/snippets/csharp/cli/cli_with_config.cs
```csharp title="cli_with_config.cs"
using System;
using System.CommandLine;
using System.Text.Json;
using System.Threading.Tasks;
using Xberg;

var rootCommand = new RootCommand("Xberg with configuration");

var extractCommand = new Command("extract", "Extract with custom configuration");
var filePath = new Argument<string>("path", "Document file path");
var configPath = new Option<string>(
    new[] { "-c", "--config" },
    "Path to JSON configuration file"
);
var forceOcr = new Option<bool>(
    new[] { "--force-ocr" },
    "Force OCR processing"
);
var useCache = new Option<bool>(
    new[] { "--use-cache" },
    getDefaultValue: () => true,
    "Use caching (default: true)"
);

extractCommand.AddArgument(filePath);
extractCommand.AddOption(configPath);
extractCommand.AddOption(forceOcr);
extractCommand.AddOption(useCache);

extractCommand.SetHandler(async (path, config, ocr, cache) =>
{
    try
    {
        ExtractionConfig extractionConfig;

        if (!string.IsNullOrEmpty(config))
        {
            var json = await System.IO.File.ReadAllTextAsync(config);
            extractionConfig = JsonSerializer.Deserialize<ExtractionConfig>(json);
        }
        else
        {
            extractionConfig = new ExtractionConfig
            {
                UseCache = cache,
                ForceOcr = ocr,
            };
        }

        Console.WriteLine("Extracting with configuration:");
        Console.WriteLine($"  - File: {path}");
        Console.WriteLine($"  - Force OCR: {extractionConfig.ForceOcr}");
        Console.WriteLine($"  - Use Cache: {extractionConfig.UseCache}");

        var result = await XbergLib.ExtractAsync(path, extractionConfig);

        Console.WriteLine($"\nExtraction complete:");
        Console.WriteLine($"  - Content length: {result.Content.Length}");
        Console.WriteLine($"  - Format: {result.Metadata.FormatType}");
        Console.WriteLine($"  - Languages: {string.Join(", ", result.DetectedLanguages)}");

        Console.WriteLine($"\n{result.Content}");
    }
    catch (Exception ex)
    {
        Console.Error.WriteLine($"Error: {ex.Message}");
        Environment.Exit(1);
    }
}, filePath, configPath, forceOcr, useCache);

rootCommand.AddCommand(extractCommand);

return await rootCommand.InvokeAsync(args);
```
